# Experiment 01

# Task 02

## Perform Data Acquisition and Exploratory Analysis Using Descriptive Statistical Measures

### Student Details

**Name:** Karthik

**Course:** B.Tech AIML

**Experiment:** 01

**Task:** 02

### Objective

To perform descriptive statistical analysis on the Iris dataset using Pandas and understand the characteristics of the dataset.

## Step 1

### Import Required Libraries

NumPy is used for numerical computations.

Pandas is used for data analysis and data manipulation.

In [1]:
import numpy as np
import pandas as pd

## Step 2

### Loading the Iris Dataset

The Iris dataset is loaded into a Pandas DataFrame using the `read_csv()` function.

In [3]:
iris = pd.read_csv("iris.csv")

iris.head()

,sepal_length,sepal_width,petal_length,petal_width,species
0,5.1,3.5,1.4,0.2,setosa
1,4.9,3.0,1.4,0.2,setosa
2,4.7,3.2,1.3,0.2,setosa
3,4.6,3.1,1.5,0.2,setosa
4,5.0,3.6,1.4,0.2,setosa


## Step 3

# Dataset Shape

The `shape` attribute returns the number of rows and columns in the dataset.

It helps us understand the size of the dataset before performing analysis.

In [4]:
print("Shape of the dataset:", iris.shape)

Shape of the dataset: (150, 5)


## Step 4

# Data Types of Each Column

The `dtypes` attribute displays the data type of each column in the dataset.

Knowing the data type helps determine which statistical operations can be performed.

In [5]:
iris.dtypes

,0
sepal_length,float64
sepal_width,float64
petal_length,float64
petal_width,float64
species,object


## Step 5

# Descriptive Statistical Summary

The `describe()` function generates summary statistics for all numerical columns.

It provides information such as:
- Count
- Mean
- Standard Deviation
- Minimum
- Maximum
- Quartiles (25%, 50%, 75%)

In [6]:
iris.describe()

,sepal_length,sepal_width,petal_length,petal_width
count,150.000000,150.000000,150.000000,150.000000
mean,5.843333,3.057333,3.758000,1.199333
std,0.828066,0.435866,1.765298,0.762238
min,4.300000,2.000000,1.000000,0.100000
25%,5.100000,2.800000,1.600000,0.300000
50%,5.800000,3.000000,4.350000,1.300000
75%,6.400000,3.300000,5.100000,1.800000
max,7.900000,4.400000,6.900000,2.500000


## Step 6

# Mean (Average)

The **mean** is the average value of a numerical column.

It is calculated by adding all values in a column and dividing by the total number of values.

The mean helps us understand the central tendency of the dataset.

In [7]:
iris.mean(numeric_only=True)

,0
sepal_length,5.843333
sepal_width,3.057333
petal_length,3.758000
petal_width,1.199333


## Step 7

# Median

The **median** is the middle value of a dataset after arranging the values in ascending order.

If the number of observations is even, the median is the average of the two middle values.

The median is less affected by outliers than the mean.

In [8]:
iris.median(numeric_only=True)

,0
sepal_length,5.80
sepal_width,3.00
petal_length,4.35
petal_width,1.30


## Step 8

# Mode

The **mode** is the value that occurs most frequently in a dataset.

A dataset may have one mode, multiple modes, or no mode.

In [10]:
iris.mode()

,sepal_length,sepal_width,petal_length,petal_width,species
0,5.0,3.0,1.4,0.2,setosa
1,NaN,NaN,1.5,NaN,versicolor
2,NaN,NaN,NaN,NaN,virginica


## Step 9

# Variance

Variance measures how far the data values are spread from the mean.

A larger variance indicates greater variability in the data.

In [12]:
iris.var(numeric_only=True)

,0
sepal_length,0.685694
sepal_width,0.189979
petal_length,3.116278
petal_width,0.581006


## Step 10

# Standard Deviation

Standard deviation is the square root of the variance.

It indicates how much the values deviate from the mean.

A low standard deviation means the values are close to the mean, while a high standard deviation indicates greater variation.

In [13]:
iris.std(numeric_only=True)

,0
sepal_length,0.828066
sepal_width,0.435866
petal_length,1.765298
petal_width,0.762238


## Step 11

# Correlation Matrix

Correlation measures the relationship between two numerical variables.

The correlation coefficient ranges from **-1 to +1**.

- **+1** → Perfect positive correlation
- **0** → No correlation
- **-1** → Perfect negative correlation

The correlation matrix helps identify relationships between different features.

In [14]:
iris.corr(numeric_only=True)

,sepal_length,sepal_width,petal_length,petal_width
sepal_length,1.000000,-0.117570,0.871754,0.817941
sepal_width,-0.117570,1.000000,-0.428440,-0.366126
petal_length,0.871754,-0.428440,1.000000,0.962865
petal_width,0.817941,-0.366126,0.962865,1.000000


## Step 12

# Checking Missing Values

Missing values can affect the accuracy of data analysis and machine learning models.

The `isnull().sum()` function counts the number of missing values in each column.

In [15]:
iris.isnull().sum()

,0
sepal_length,0
sepal_width,0
petal_length,0
petal_width,0
species,0


## Step 13

# Finding Unique Values

The `nunique()` function returns the number of unique values in each column.

It helps understand the diversity of values present in the dataset.

In [16]:
iris.nunique()

,0
sepal_length,35
sepal_width,23
petal_length,43
petal_width,22
species,3


## Step 14

# Counting Species

The `value_counts()` function counts the occurrences of each category in the dataset.

It is useful for understanding the distribution of categorical variables.

In [17]:
iris["species"].value_counts()

,count
species,
setosa,50
versicolor,50
virginica,50


In [18]:
# =====================================================
# Experiment 01 - Task 02
# Descriptive Statistical Analysis on Iris Dataset
# =====================================================

# Import Libraries
import numpy as np
import pandas as pd

# Load Dataset
iris = pd.read_csv("iris.csv")

# Display First Five Rows
print("========== First Five Rows ==========")
print(iris.head())

# Display Last Five Rows
print("\n========== Last Five Rows ==========")
print(iris.tail())

# Dataset Shape
print("\n========== Dataset Shape ==========")
print("Rows and Columns:", iris.shape)

# Data Types
print("\n========== Data Types ==========")
print(iris.dtypes)

# Column Names
print("\n========== Column Names ==========")
print(iris.columns)

# Statistical Summary
print("\n========== Descriptive Statistics ==========")
print(iris.describe())

# Mean
print("\n========== Mean ==========")
print(iris.mean(numeric_only=True))

# Median
print("\n========== Median ==========")
print(iris.median(numeric_only=True))

# Mode
print("\n========== Mode ==========")
print(iris.mode())

# Variance
print("\n========== Variance ==========")
print(iris.var(numeric_only=True))

# Standard Deviation
print("\n========== Standard Deviation ==========")
print(iris.std(numeric_only=True))

# Correlation Matrix
print("\n========== Correlation Matrix ==========")
print(iris.corr(numeric_only=True))

# Missing Values
print("\n========== Missing Values ==========")
print(iris.isnull().sum())

# Number of Unique Values
print("\n========== Unique Values ==========")
print(iris.nunique())

# Species Count
print("\n========== Species Count ==========")
print(iris["species"].value_counts())

print("\n========== Task 2 Completed Successfully ==========")

========== First Five Rows ==========
   sepal_length  sepal_width  petal_length  petal_width species
0           5.1          3.5           1.4          0.2  setosa
1           4.9          3.0           1.4          0.2  setosa
2           4.7          3.2           1.3          0.2  setosa
3           4.6          3.1           1.5          0.2  setosa
4           5.0          3.6           1.4          0.2  setosa

========== Last Five Rows ==========
     sepal_length  sepal_width  petal_length  petal_width    species
145           6.7          3.0           5.2          2.3  virginica
146           6.3          2.5           5.0          1.9  virginica
147           6.5          3.0           5.2          2.0  virginica
148           6.2          3.4           5.4          2.3  virginica
149           5.9          3.0           5.1          1.8  virginica

========== Dataset Shape ==========
Rows and Columns: (150, 5)

========== Data Types ==========
sepal_length    float64
sepa

# Conclusion

Successfully performed descriptive statistical analysis on the Iris dataset.

The dataset consists of **150 observations** and **5 attributes**. Statistical measures such as mean, median, mode, variance, standard deviation, and correlation were calculated to understand the characteristics of the dataset.

The dataset contains **no missing values**, and all three species are equally represented, making it suitable for exploratory data analysis and machine learning tasks.